packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -257,13 +301,75 @@ export const fetchIndexes = (
     }
   };
 };
-
 export const refreshRegularIndexes = (): IndexesThunkAction<
   Promise<void>,
   FetchIndexesActions
+> => {
+  return async (dispatch, getState) => {
+    const { status } = getState().regularIndexes;
+
+    // If we are in a READY state, then we have already fetched the data
+    // and are refreshing the list.
+    const newStatus: FetchStatus =
+      status === FetchStatuses.READY
+        ? FetchStatuses.REFRESHING
+        : FetchStatuses.FETCHING;
```

Comment:
Kinda feels like it mixes responsibilites of the action too much: the "state machine" job that's usually should be the responsibility of the reducer is done by the action now, additionally from the reducer perspective if would make one action to look like a different one: like imagine you're debugging an issue with the indexes feching and inspecing the log of actions passing through the store, you now can't track the action to the exact location it was dispatched from because refresh action creator will also emit a fetch event that is usually coming from a different place in the code

Reply:
I kinda just copied that from search indexes to make it work the same and didn't think too much of it. Should I just move this into fetchIndexes() or are you suggesting a bigger refactor?

Reply:
Moving it to fetch indexes would kinda have the same issue: action doing the work of the reducer, it is better in terms of at least it's sort of just one place now, but you're still loosing information about what actually happened in the application. The easiest way to align with style guide here would be to pass what type of action triggered the fetch and then reducer to make a decision about the resulting state based on that: so like the action type would have something like `trigger: 'poll' | 'refresh' | 'initial-fetch'` instead of a new status as prop. That way we're making it clear what was the trigger and remove the "avoid using actions as setters" issue

Reply:
I got the concept of a "fetch reason" now which then gets mapped to status in the reducer.